modules/text_server_adv/text_server_adv.cpp

```diff
@@ -1419,25 +1419,17 @@ _FORCE_INLINE_ bool TextServerAdvanced::_ensure_cache_for_size(FontAdvanced *p_f
 			fargs.flags = FT_OPEN_MEMORY;
 			fargs.stream = &fd->stream;
 
-			int max_index = 0;
-			FT_Face tmp_face = nullptr;
-			error = FT_Open_Face(ft_library, &fargs, -1, &tmp_face);
-			if (tmp_face && error == 0) {
-				max_index = tmp_face->num_faces - 1;
-			}
-			if (tmp_face) {
-				FT_Done_Face(tmp_face);
-			}
-
-			error = FT_Open_Face(ft_library, &fargs, CLAMP(p_font_data->face_index, 0, max_index), &fd->face);
+			error = FT_Open_Face(ft_library, &fargs, p_font_data->face_index, &fd->face);
```

Comment:
I cannot find in the [freetype documentation](https://freetype.org/freetype2/docs/reference/ft2-face_creation.html#ft_open_face) if giving an index that is out of bounds (more than `tmp_face->num_faces` returns as an upper bound) is clamped or returns an error. Did you try it ? And maybe add it as a comment above (like this can fail if face_index is wrong somehow) ?

Reply:
I tried it, yes. It simply returns with an error if the index is out of bounds.

Reply:
Then maybe a quick comment above would make it clearer.

Reply:
I did add the font face index to the error message that's shown to the user, so it's clearer that it may have had something to do with that.
Other than that, well, it also fails gracefully if any other parts of the font input are wrong. I'm not sure I agree we need to specifically note also fails if the face index is incorrect.

Reply:
It was just to make it clear as this is the only one were you have control over. But as you added it in the error message, you are right, it's probably fine like this.